<robot name="scanner">

  <link name="lidar_origin" >
    <visual>
      <geometry>
        <mesh filename="package://rplidar_ros/meshes/support.STL" scale = "0.001 0.001 0.001" />
      </geometry>
      <origin xyz="-0.158 -0.51 0" rpy="0 0 0"/>
    </visual>
  </link>
  <link name="lidar_platform" >
    <visual>
      <geometry>
        <mesh filename="package://rplidar_ros/meshes/platform.STL" scale = "0.001 0.001 0.001" />
      </geometry>
      <origin xyz="-0.091 -0.02 -0.05" rpy="0 0 0"/>
    </visual>
  </link>

  <joint name="rail" type="prismatic">
    <parent link="lidar_origin"/>
    <child link="lidar_platform"/>
    <limit effort="1000.0" lower="-0.38" upper="0" velocity="0.5"/>
  </joint>

</robot>
